finch/src/services/__tests__/logger.test.js

```diff
@@ -0,0 +1,141 @@
+import config from "../../../config/index.js";
+import { logInfo, logError, logWarning, formatLog } from "../logger";
+
+
+/* eslint-disable no-console */
+describe('Services/logger', () => {
+  describe('#logInfo', () => {
+    beforeEach(() => {
+      /* eslint-disable-next-line no-console */
+      console.log = jest.fn();
+    });
+
+    test('should call console.log with correct arguments when prettify true', () => {
+      config.logs.prettify = true;
+      logInfo('testing msg for log info', {
+        someKey: 'someValue',
+      });
+      expect(console.log).toHaveBeenCalledTimes(1);
+      expect(console.log).toHaveBeenCalledWith({
+        msg: 'testing msg for log info',
+        type: 'info',
+        context: {
+          someKey: 'someValue',
+        },
+      });
+    });
+
+    test('should call console.log with stringified params when prettify false', () => {
+      config.logs.prettify = false;
+      logInfo('testing msg for log info', {
+        someKey: 'someValue',
+      });
+      expect(console.log).toHaveBeenCalledTimes(1);
+      expect(console.log).toHaveBeenCalledWith(JSON.stringify({
+        type: 'info',
+        msg: 'testing msg for log info',
+        context: {
+          someKey: 'someValue',
+        },
+      }));
+    });
+  });
+
+  describe('#logError', () => {
+    beforeEach(() => {
+      /* eslint-disable-next-line no-console */
+      console.error = jest.fn();
+    });
+
+    test('should call console.error with correct arguments when prettify true', () => {
+      config.logs.prettify = true;
+      logError('testing msg for error', {
+        someKey: 'someValue',
+      });
+      expect(console.error).toHaveBeenCalledTimes(1);
+      expect(console.error).toHaveBeenCalledWith({
+        msg: 'testing msg for error',
+        type: 'error',
+        context: {
+          someKey: 'someValue',
+        },
+      });
+    });
+
+    test('should call console.error with stringified params when prettify false', () => {
+      config.logs.prettify = false;
+      logError('testing msg for error', {
+        someKey: 'someValue',
+      });
+      expect(console.error).toHaveBeenCalledTimes(1);
+      expect(console.error).toHaveBeenCalledWith(JSON.stringify({
+        type: 'error',
+        msg: 'testing msg for error',
+        context: {
+          someKey: 'someValue',
+        },
+      }));
+    });
+  });
+
+  describe('#logWarn', () => {
+    beforeEach(() => {
```

Comment:
Should there be a corresponding `afterEach` to prevent forever stubbing the method? Sorry if it's already there somewhere. I might have missed it.

Reply:
actually it's not there, but I didn't see any problem happening so I thought there is no problem happening, but will check & come back to you